interface/src/app/signature/signature.js

```diff
@@ -37,56 +68,66 @@ angular.module('adage.signature', [
       function success(response) {
         self.name = response.name;
         self.mlmodel = response.mlmodel.title;
+        self.organism = response.mlmodel.organism.scientific_name;
         self.statusMessage = '';
       },
-      function error(err) {
-        $log.error('Failed to get signature information: ' + err.statusText);
-        self.statusMessage = 'Failed to get signature information from server';
+      function error(response) {
+        var errMessage = errGen('Failed to get signature information: ',
+                                response);
+        $log.error(errMessage);
+        self.statusMessage = errMessage + '. Please try again later.';
       }
     );
-    self.organism = 'Pseudomonas aeruginosa';
     self.genes = [];
   }
 ])
 
-.directive('highWeightGenes', ['Participation', '$log',
+.directive('participatoryGenes', ['Participation', '$log',
```

Comment:
I noticed that you used `errGen` service later. Can you add it in dependency list?